lib/open_api_spex/cast_parameters.ex

```diff
@@ -7,39 +7,94 @@ defmodule OpenApiSpex.CastParameters do
   @spec cast(Plug.Conn.t(), Operation.t(), Components.t()) ::
           {:error, [Error.t()]} | {:ok, Conn.t()}
   def cast(conn, operation, components) do
-    # Taken together as a set, operation parameters are similar to an object schema type.
-    # Convert parameters to an object schema, then delegate to `Cast.Object.cast/1`
-
-    # Operation's parameters list may include references - resolving here
-    resolved_parameters =
-      Enum.map(operation.parameters, fn
-        ref = %Reference{} -> Reference.resolve_parameter(ref, components.parameters)
-        param = %Parameter{} -> param
-      end)
+    full_cast_result =
```

Comment:
One more refactor I just thought of: This expression can be extracted to its own private function called `cast_to_params()`. Then the `case` statement below can be converted to a `with`, that includes the new function call. That will make this function much smaller.

Reply:
Great idea, I've just copied what you've said :smile: